cmake/BLTMacros.cmake

```diff
@@ -128,10 +128,71 @@ endmacro(blt_add_target_definitions)
 
 
 ##------------------------------------------------------------------------------
+## blt_add_target_compile_flags (TO <target> FLAGS [FOO [BAR ...]])
+##
+## Adds compiler flags to a target by appending to the target's existing flags.
+##------------------------------------------------------------------------------
+macro(blt_add_target_compile_flags)
```

Comment:
Do we want to support prepending flags as well? I would have append be the default but add a PREPEND option